sklearn/utils/tests/test_utils.py

```diff
@@ -464,6 +464,17 @@ def test_safe_indexing_container_axis_0_unsupported_type():
         _safe_indexing(array, indices, axis=0)
 
 
+def test_safe_indexing_pandas_no_settingwithcopy_warning():
+    # Using safe_indexing with an array-like indexer gives a copy of the
+    # DataFrame -> ensure it doesn't raise a warning if modified
+    pd = pytest.importorskip("pandas")
+    X = pd.DataFrame({"a": [1, 2, 3], "b": [3, 4, 5]})
+    subset = _safe_indexing(X, [0, 1], axis=0)
+    with pytest.warns(None) as record:
+        subset.iloc[0, 0] = 10
+    assert len(record) == 0
```

Comment:
Could you print the warning message such that we can see why one of the CI is failing?

Reply:
Depending on the type of warnings raised and since we import pandas, we could check that `SettingWithCopyWarning` is not raised.

Reply:
Ah, that might be for an older version of pandas (the failing build is using debian with pandas 0.23)

Reply:
Yes, I think skipping for older pandas versions is the easiest (in theory, we could pass a keyword argument to `take` to get the correct behaviour for pandas < 1.0, but that keyword is deprecated for newer versions, so then we would need version dependent logic in the code itself, which doesn't seem worth it)

Reply:
What makes me think that it is not worth is that we actually officially require `pandas >= 0.25` normally:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/_min_dependencies.py#L30

Our CI is indeed testing 0.23.3.

Reply:
Feel free to upgrade the CI config as part of this PR if you wish.